modules/travel_pay/app/services/travel_pay/claims_service.rb

```diff
@@ -160,6 +169,57 @@ def get_document_summaries(veis_token, btsss_token, claim_id)
       documents
     end
 
+    def find_decision_letter_document(claim)
+      return nil unless claim&.dig('documents')
+
+      claim['documents'].find do |document|
+        filename = document['filename'] || ''
+        filename.match?(/Decision Letter|Rejection Letter/i)
+      end
+    end
+
+    def get_decision_reason(claim_id, document_id)
+      documents_service = TravelPay::DocumentsService.new(@auth_manager)
+      document_data = documents_service.download_document(claim_id, document_id)
```

Comment:
This feature should not bubble up errors to the controller. 

We should 
1. Verify errors bubbled up from this don't break all of claim details
2. Wrap this call in a `begin/rescue` for safety.